pimdm/Config.py

```diff
@@ -34,18 +34,25 @@ def parse_config_file(file_path):
             pim_globals.T_LIMIT = pim_config["DefaultTimers"].get("T_LIMIT", pim_globals.T_LIMIT)
 
         if "Interfaces" in pim_config:
-            interfaces = pim_config["Interfaces"]  # type: dict
-
-            for if_name, if_value in interfaces.items():
-                if if_value.get("ipv4", False):
-                    if_ipv4 = if_value["ipv4"]
-                    if if_ipv4.get("enabled", False):
-                        Main.add_pim_interface(interface_name=if_name, state_refresh_capable=if_ipv4.get("state_refresh", False), ipv4=True, ipv6=False)
-
-                if if_value.get("ipv6", False):
-                    if_ipv6 = if_value["ipv6"]
-                    if if_ipv6.get("enabled", False):
-                        Main.add_pim_interface(interface_name=if_name, state_refresh_capable=if_ipv6.get("state_refresh", False), ipv4=False, ipv6=True)
+            interface_dict = pim_config["Interfaces"]
+            failing_if_list = []
+
+            for if_name, if_value in interface_dict.items():
+                try:
+                    if if_value.get("ipv4", False):
+                        if_ipv4 = if_value["ipv4"]
+                        if if_ipv4.get("enabled", False):
+                            Main.add_pim_interface(interface_name=if_name, state_refresh_capable=if_ipv4.get("state_refresh", False), ipv4=True, ipv6=False)
+
+                    if if_value.get("ipv6", False):
+                        if_ipv6 = if_value["ipv6"]
+                        if if_ipv6.get("enabled", False):
+                            Main.add_pim_interface(interface_name=if_name, state_refresh_capable=if_ipv6.get("state_refresh", False), ipv4=False, ipv6=True)
+                except Exception as e:
+                    print(e, file=sys.stderr)
+                    failing_if_list.append(if_name)
+            for if_name in failing_if_list:
+                del interface_dict[if_name]
```

Comment:
Why delete such interfaces if the dict variable is then replaced by IGMP and MLD dict references?

Reply:
Right as the dictionary is overridden it does not make any sens.